cpp/src/gandiva/precompiled/time.cc

```diff
@@ -15,8 +15,45 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <sstream>
 #include "./epoch_time_point.h"
 
+std::string ensure_two_digits(int64_t num) {
+  std::stringstream s;
+  if (num < 10) {
+    s << "0";
+  }
+  s << num;
+  return s.str();
+}
+
+std::string ensure_three_digits(int64_t num) {
+  std::stringstream s;
+  if (num < 10) {
+    s << "0";
+  }
+  if (num < 100) {
+    s << "0";
+  }
+  s << num;
+  return s.str();
+}
+
+std::string ensure_four_digits(int64_t num) {
```

Comment:
you can use sprint/snprintf. 

https://www.quora.com/What-is-the-difference-between-2d-and-02d-in-C-language-What-is-the-logic

Reply:
done